tasks/internal_reconcile_report_phantom/internal_reconcile_report_phantom.py

```diff
@@ -0,0 +1,204 @@
+
+PAGE_SIZE = 100
+
+
+def task(
+    job_id: int,
+    page_index: int,
+    db_connect_info: Dict[str, str],
+) -> Dict[str, Any]:
+    """
+    Args:
+        job_id: The unique ID of job/report.
+        page_index: The 0-based index of the results page to return.
+        db_connect_info: See requests_db.py's get_configuration for further details.
+    """
+    engine = shared_db.get_user_connection(db_connect_info)
+    granules = query_db(
+        engine,
+        job_id,
+        page_index,
+    )
+
+    return {
+        OUTPUT_JOB_ID_KEY: job_id,
+        OUTPUT_ANOTHER_PAGE_KEY: len(granules) > PAGE_SIZE,
+        OUTPUT_PHANTOMS_KEY: granules[0:PAGE_SIZE],  # we get one extra for anotherPage calculation.
+    }
+
+
+@retry_operational_error()
+def query_db(
+    engine: Engine,
+    job_id: str,
+    page_index: int,
+) -> List[Dict[str, Any]]:
+    """
+    Gets phantoms for the given job/page, up to PAGE_SIZE + 1 results.
+
+    Args:
+        engine: The sqlalchemy engine to use for contacting the database.
+        job_id: The unique ID of job/report.
+        page_index: The 0-based index of the results page to return.
+
+    Returns:
+        A list containing dicts matching the format of "phantoms" in output.json.
+    """
+    LOGGER.info(f"Retrieving page '{page_index}' of reports for job '{job_id}'")
+    with engine.begin() as connection:
+        sql_results = connection.execute(
+            get_phantoms_sql(),
+            [
+                {
+                    "job_id": job_id,
+                    "page_index": page_index,
+                    "page_size": PAGE_SIZE,
+                }
+            ],
+        )
+
+        phantoms = []
+        for sql_result in sql_results:
+            phantoms.append(
+                {
+                    PHANTOMS_COLLECTION_ID_KEY: sql_result["collection_id"],
+                    PHANTOMS_GRANULE_ID_KEY: sql_result["granule_id"],
+                    PHANTOMS_FILENAME_KEY: sql_result["filename"],
+                    PHANTOMS_KEY_PATH_KEY: sql_result["key_path"],
+                    PHANTOMS_ORCA_ETAG_KEY: sql_result["orca_etag"],
+                    PHANTOMS_ORCA_LAST_UPDATE_KEY: sql_result["orca_last_update"],
+                    PHANTOMS_ORCA_SIZE_KEY: sql_result["orca_size"],
+                }
+            )
+        return phantoms
+
+
+def get_phantoms_sql() -> text:
+    """
+    SQL for getting phantom report entries for a given job_id, page_size, and page_index.
+    Formats datetimes in milliseconds since 1 January 1970 UTC.
+    """
+    return text(
+        """
+SELECT
+    collection_id, 
+    granule_id, 
+    filename, 
+    key_path, 
+    orca_etag, 
+    (EXTRACT(EPOCH FROM date_trunc('milliseconds', orca_last_update) AT TIME ZONE 'UTC') * 1000)::bigint as orca_last_update,
+    orca_size
+    FROM reconcile_phantom_report
+    WHERE job_id = :job_id
+    ORDER BY collection_id, granule_id, filename
+    OFFSET :page_index*:page_size
```

Comment:
For our initial prototype, this type of paging will work but, it is not performant and as the record size increases this will cause additional slowness. See https://bugs.earthdata.nasa.gov/browse/ORCA-286 as a reference. Ideally, we would want to use a cursor based approach possibly based on collection and granule id. See the following references for the schema to see what a potential schema for cursor and implementation.
- https://relay.dev/graphql/connections.htm#sec-undefined.PageInfo.Fields
- https://graphql.org/learn/pagination/
- https://slack.engineering/evolving-api-pagination-at-slack/
- https://medium.com/swlh/how-to-implement-cursor-pagination-like-a-pro-513140b65f32

Consider creating a performance card to revisit paging like ORCA-286.